### PR TITLE
Bug 1833093 - Add a Nimbus experiment for pdf.js

### PR DESCRIPTION
**Pull request by calixteman**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1833093

### CODE REVIEW

File: android-components/components/browser/engine-gecko/build.gradle
Comment (Amejia481) [inline android-components/components/browser/engine-gecko/build.gradle:100]: Should we also have nightly channel?

File: android-components/components/browser/engine-gecko/build.gradle
Comment (Amejia481) [inline android-components/components/browser/engine-gecko/build.gradle:96]: I wonder if we should have `nimbus.fml.yaml`, which applies to the whole AC and not only to the `engine-gecko` module, @jhugman what do you think?

File: fenix/app/nimbus.fml.yaml
Comment (Amejia481) [inline fenix/app/nimbus.fml.yaml:141]: Should we add a `default` property for other channels?

File: android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt
Comment (calixteman) [reply android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt:1102]: I added this method in GeckoSession.java:
https://phabricator.services.mozilla.com/D178045#C6430005NL3628

Comment (Amejia481) [reply android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt:1102]: 👍🏽

File: android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt
Comment (jhugman) [inline android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt:1116]: Note to future: [this PR](https://github.com/mozilla/application-services/pull/5574) adds `getFeature(featureId: String)` and `toJSONObject()` for usage here.

File: android-components/components/browser/engine-gecko/build.gradle
Comment (jhugman) [reply android-components/components/browser/engine-gecko/build.gradle:96]: No, for a number of reasons:

- the generated code is per app, and not every app needs every component. You could use a FML component per app, and share FML files by including them in to each of the app's component, however…
- the generated code needs to be accessible from the component itself. If you had a per-app FML component, then the components (e.g. `engine-gecko`) would need to know which app it's in at compile time.

I have thought about having a per app FML file in AC, which is just `imports` of the component FML files, which then gets included by the apps: this might be work; I'd suggest thinking about this refactor when the third component grows an FML file.

File: android-components/components/browser/engine-gecko/build.gradle
Comment (jhugman) [reply android-components/components/browser/engine-gecko/build.gradle:100]: Imported channels are less useful, and more confusing than I thought they were going to be. We may need to add channels here, depending on how the build is setup.

**Review by jhugman — APPROVED** [review]
Looking good so far! You may want to hold off for [`toJSONObject` to land](https://github.com/mozilla/application-services/pull/5574) before landing this, or not.

Approving to keep you unblocked.

File: android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt
Comment (calixteman) [reply android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt:1104]: I updated the patch to use this amazing new feature: @jhugman thank you very much.

**Comment by Amejia481** [comment]
While testing I noticed by default in Fenix/Focus the features are disabled by default,  we'll need to investigate this further, @jhugman do you have an idea of why this could be happening ? Noticing we are already setting the default values to be `true` on `geckoview.fml.yaml` but for some reason we are getting `false`.

**Comment by Amejia481** [comment]
Thanks @calixteman for letting my know offline that I have to remove the `FxNimbus.kt` file to see the feature changes!

File: android-components/components/browser/engine-gecko/build.gradle
Comment (badboy) [inline android-components/components/browser/engine-gecko/build.gradle:60]: What about this here:

```suggestion
    implementation(project(':service-nimbus')) {
        exclude group: 'org.mozilla.telemetry', module: 'glean-native'
    }
```

The issue is (I think) that engine-gecko depends on geckoview, which itself provides the glean-native part.
But then nimbus also depends on `glean-native` transitively, but doesn't know about geckoview, so it's pulled in again.
By excluding that explicitly it should fall back to the geckoview-provided one when it's all compiled together.

I ran `./gradlew lint` on this change locally and it seemed to work, but I didn't run any tests.
We do something similar here: https://github.com/mozilla-mobile/firefox-android/blob/8dc9f859a4f75050c2d5f6b020a54963211aadcf/android-components/components/service/sync-logins/build.gradle#L35-L37

**Pull request merged**
